test/integration/framework/master_utils.go

```diff
@@ -190,8 +190,9 @@ func startMasterOrDie(masterConfig *master.Config, incomingServer *httptest.Serv
 	masterConfig.ExtraConfig.VersionedInformers = informers.NewSharedInformerFactory(clientset, masterConfig.GenericConfig.LoopbackClientConfig.Timeout)
 	m, err = masterConfig.Complete().New(genericapiserver.NewEmptyDelegate())
 	if err != nil {
+		defer klog.Fatalf("error in bringing up the master: %v", err)
 		closeFn()
-		klog.Fatalf("error in bringing up the master: %v", err)
+		return m, s, closeFn
```

Comment:
Is the return statement necessary here?

Reply:
Without the return statement, wouldn't execution continue beyond the if block ?
Whereas the previous code would panic at the conclusion of the if block.

Reply:
That makes sense, wonder if there is a cleaner way of handling this. Since we want to 
@misterikkit ideas?

Reply:
What is it about `closeFn` that it can crash sometimes? (would be nice to have a more detailed commit message explaining the reasoning.)

I think this would be cleaner if closeFn() returned an error that we checked, then we could gracefully log both errors and exit. That's obviously a bigger change.

Anyway, down on line 227 it looks like we're doing a similar thing of logging an error and calling closeFn. Maybe this can be done like that?

Reply:
@misterikkit 
Maybe you can clarify your suggestion w.r.t. code on line 227:
```
		closeFn()
		klog.Errorf("last health content: %q", string(lastHealthContent))
		klog.Fatal(err)
```
If closeFn() crashes, we wouldn't see the two subsequent errors.

To my understanding, the current formation allows the error to be shown even if closeFn() crashes - without duplicating the error shown.

Reply:
@misterikkit @saad-ali 
I can change the return statement to:
```
		return nil, nil, closeFn
```
if you prefer the above version.

Reply:
@misterikkit 
I have updated description.